docs/spec/consensus/fork-accountability.md

```diff
@@ -0,0 +1,320 @@
+
+
+Independently of a fork happening, punishing this behavior might be important to prevent forks altogether. This should keep attackers from misbehaving: if at most 1/3 of the voting power is faulty, this misbehavior is detectable but will not lead to a safety violation. Thus, unless they have more than 1/3 (or in some cases more than 2/3) of the voting power attackers have the incentive to not misbehave. If attackers control too much voting power, we have to deal with forks, as discussed in this document.   
+
+
+## Two types of forks
+
+* Fork-Full. Two correct validators decide on different blocks for the same height. Since also the next validator sets are decided upon, the correct validators may be partitioned to participate in two distinct branches of the forked chain.
+
+As in this case we have two different blocks (both having the same right/no right to exist), a central system invariant (one block per height decided by correct validators) is violated. As full nodes are contaminated in this case, the contamination can spread also to lite clients. However, even without breaking this system invariant, lite clients can be subject to a fork:
+
+* Fork-Lite. All correct validators decide on the same block for height *h*, but faulty processes (validators or not), forge a different block for that height, in order to fool users (who use the lite client).
+
+
+# Attack scenarios
+
+## On-chain attacks
+
+### Equivocation (one round)
+
+There are several scenarios in which forks might happen. The first is double signing within a round.
+
+* F1. Equivocation: faulty validators sign multiple vote messages (prevote and/or precommit) for different values *during the same round r* at a given height h.
+
+
+### Flip-flopping
+
+Tendermint consensus implements a locking mechanism: If a correct validator *p* receives *2f + 1* prevotes for a value *v* in round *r*, it locks *v* and remembers *r*. In this case, *p* also sends a precommit message for *v*, which later may serve as proof that *p* signed *v*.
+In subsequent rounds, *p* only sends prevote messages for a value it had previously locked. However, it is possible to change the locked value if in a future round *r' > r*, if the process receives *2f + 1* prevotes for a different value *v'*. In this case, *p* sends a precommit for *v'*. This algorithmic feature can be exploited in two ways:
+
+
+
+* F2. Faulty Flip-flopping (Amnesia): faulty validators precommit some value *v* in round *r* (value *v* is locked in round *r*) and then prevote for different value *v'* in higher round *r' > r* without previously correctly unlocking value *v*. In this case faulty processes "forget" that they have locked value *v* and prevote some other value in the following rounds.
+Some correct validators might have decided on *v* in *r*, and other correct validators decide on *v'* in *r'*. Here we can have branching on the main chain (Fork-Full).
```

Comment:
Amnesia can also be used to attack lite clients without fork on the main chain. In that case, faulty validators after they have decided in round r, keep signing messages in higher rounds for different values, but send those messages only to lite clients.